Src/FluentAssertions/Primitives/DateOnlyAssertions.cs

```diff
@@ -0,0 +1,527 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using FluentAssertions.Execution;
+
+#if NET6_0_OR_GREATER
+
+namespace FluentAssertions.Primitives
+{
+    /// <summary>
+    /// Contains a number of methods to assert that a <see cref="DateOnly"/> is in the expected state.
+    /// </summary>
+    /// /// <remarks>
+    /// You can use the <see cref="FluentAssertions.Extensions.FluentDateOnlyExtensions"/>
+    /// for a more fluent way of specifying a <see cref="DateOnly"/>.
+    /// </remarks>
+    [DebuggerNonUserCode]
+    public class DateOnlyAssertions : DateOnlyAssertions<DateOnlyAssertions>
+    {
+        public DateOnlyAssertions(DateOnly? value)
+            : base(value)
+        {
+        }
+    }
+
+#pragma warning disable CS0659 // Ignore not overriding Object.GetHashCode()
+#pragma warning disable CA1065 // Ignore throwing NotSupportedException from Equals
+    /// <summary>
+    /// Contains a number of methods to assert that a <see cref="DateOnly"/> is in the expected state.
+    /// </summary>
+    /// /// <remarks>
+    /// You can use the <see cref="FluentAssertions.Extensions.FluentDateOnlyExtensions"/>
+    /// for a more fluent way of specifying a <see cref="DateOnly"/>.
+    /// </remarks>
+    [DebuggerNonUserCode]
+    public class DateOnlyAssertions<TAssertions>
+        where TAssertions : DateOnlyAssertions<TAssertions>
+    {
+        public DateOnlyAssertions(DateOnly? value)
+        {
+            Subject = value;
+        }
+
+        /// <summary>
+        /// Gets the object which value is being asserted.
+        /// </summary>
+        public DateOnly? Subject { get; }
+
+        /// <summary>
+        /// Asserts that the current <see cref="DateOnly"/> is exactly equal to the <paramref name="expected"/> value.
+        /// </summary>
+        /// <param name="expected">The expected value</param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+        /// </param>
+        public AndConstraint<TAssertions> Be(DateOnly expected, string because = "", params object[] becauseArgs)
```

Comment:
🤔 Would it make sense to reuse the `DateTimeAssertions` with additional overloads and change the `Should(this DateOnly`) to reuse `DateTimeAssertions` by casting the `DateOnly` to `DateTime`?

Reply:
This will lead to failed assertions printing the "time part" of the `DateTime` which is always irrelevant for `DateOnly` structs.

By "additional overloads" do you mean telling the different methods whether to print "date and time" or only "date"?

Reply:
Other reasons why not to reuse `DateTimeAssertions` (directly):
 
 `dateOnly.Should().Subject`  would return a `DateTime` instead of a `DateOnly`.
 
 Assertions on `DateTimeAssertions` that are meaningless to `DateOnly`

* `HaveSecond`/`HaveMinute`/`HaveHour`
* `BeIn(DateTimeKind)`
* `BeSameDateAs` (this is only irrelevant as it is identical to `Be`)
* `BeCloseTo(TimeSpan)`

Reply:
> This will lead to failed assertions printing the "time part" of the DateTime which is always irrelevant for DateOnly structs.
> Other reasons why not to reuse DateTimeAssertions (directly):

Fair enough

Reply:
But thanks for bringing up the point to see if it was a feasible path.